controller_manager/include/controller_manager/controller_spec.hpp

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+/*
+ * Author: Wim Meeussen
+ */
+
+#ifndef CONTROLLER_MANAGER__CONTROLLER_SPEC_HPP_
+#define CONTROLLER_MANAGER__CONTROLLER_SPEC_HPP_
+
+#include <map>
+#include <string>
+#include <vector>
+#include "controller_interface/controller_interface.hpp"
+#include "hardware_interface/controller_info.hpp"
```

Comment:
To be honest, I initially put it in controller_manager for that same reason but found someone depending on it. 

But I cannot find that now, I'm not at my main pc. I'll take a closer look tomorrow.

Reply:
I didn't put it here because in ROS1 control it's used in RobotHW:
https://github.com/ros-controls/ros_control/blob/noetic-devel/hardware_interface/include/hardware_interface/robot_hw.h#L38
https://github.com/ros-controls/ros_control/blob/noetic-devel/hardware_interface/include/hardware_interface/robot_hw.h#L84

Do you believe this will still be used there for conflict checks?

Reply:
I'd consider this controller_manager specific information and we should keep it here for the reason of conflict checks. We may create these structs from some classes provided in hardware_interface but I don't see a problem with the controller_manager operating on this view of controller information.